test/topology_custom/test_read_repair.py

```diff
@@ -0,0 +1,127 @@
+
+from test.pylib.manager_client import ManagerClient
+from test.topology.util import wait_for_token_ring_and_group0_consistency
+
+
+logger = logging.getLogger(__name__)
+
+
+@contextlib.asynccontextmanager
+async def stopped_node(manager, node):
+    await manager.server_stop_gracefully(node.server_id)
+    try:
+        yield
+    finally:
+        await manager.server_start(node.server_id, wait_others = 1)
+
+
+@contextlib.asynccontextmanager
+async def connection(manager, node):
+    try:
+        c = Cluster([node.ip_addr])
+        s = c.connect()
+        yield s
+    finally:
+        s.shutdown()
+
+
+incremental_repair_row_tombstone_test_data = pytest.param("pk int, ck int, v int, PRIMARY KEY (pk, ck)",
+                                                          "DELETE FROM ks.tbl WHERE pk = 0 AND ck = ?",
+                                                          "INSERT INTO ks.tbl (pk, ck, v) VALUES (0, {}, 1)",
+                                                          "SELECT * FROM ks.tbl WHERE pk = 0",
+                                                          {"pk": 0, "ck": 0, "v": 1},
+                                                          {"pk": 0, "ck": 1000, "v": 1},
+                                                          id="row-tombstone")
+incremental_repair_partition_tombstone_test_data = pytest.param("pk int PRIMARY KEY, v int",
+                                                                "DELETE FROM ks.tbl WHERE pk = ?",
+                                                                "INSERT INTO ks.tbl (pk, v) VALUES ({}, 1)",
+                                                                "SELECT * FROM ks.tbl",
+                                                                {"pk": 0, "v": 1},
+                                                                {"pk": 1000, "v": 1},
+                                                                id="partition-tombstone")
+incremental_repair_test_data = [incremental_repair_row_tombstone_test_data,
+                                incremental_repair_partition_tombstone_test_data]
+
+
+@pytest.mark.parametrize("columns_spec,delete_query,insert_query,select_query,row0,row1000", incremental_repair_test_data)
+@pytest.mark.asyncio
+async def test_incremental_read_repair(columns_spec, delete_query, insert_query, select_query, row0, row1000, manager: ManagerClient):
+    """Stress the incremental read repair logic
+
+    Write a long stream of row tombstones, with a live row before and after.
+    """
+    cmdline = ["--hinted-handoff-enabled", "0", "--query-tombstone-page-limit", "1000"]
+    node1 = await manager.server_add(cmdline=cmdline)
+    node2 = await manager.server_add(cmdline=cmdline)
+
+    await wait_for_token_ring_and_group0_consistency(manager, time.time() + 30)
+
+    cql = manager.get_cql()
+
+    cql.execute("CREATE KEYSPACE ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 2}")
```

Comment:
nit: we are trying to deprecate SimpleStrategy, would be nice to move away from it in new tests already.